### PR TITLE
[Merged by Bors] - feat: basic results about `Finset.sym2`

### PR DESCRIPTION
**Pull request by eric-wieser**
---

- [x] depends on: #14282

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Sym.lean [hunk 1]
```diff
@@ -112,6 +118,67 @@ protected theorem Nodup.sym2 {xs : List α} (h : xs.Nodup) : xs.sym2.Nodup := by
...
 
+theorem map_mk_sublist_sym2 (x : α) (xs : List α) (h : x ∈ xs) :
+    map (fun y ↦ s(x, y)) xs <+ xs.sym2 := by
+  induction xs with
+  | nil => simp
+  | cons x' xs ih =>
+    simp [List.sym2]
+    cases h with
+    | head =>
+      exact Sublist.cons₂ _ (sublist_append_left _ _)
```
Comment (YaelDillies) [inline Mathlib/Data/List/Sym.lean:129]: nit: Does dot notation work here?

File: Mathlib/Data/List/Sym.lean [hunk 2]
```diff
@@ -112,6 +118,67 @@ protected theorem Nodup.sym2 {xs : List α} (h : xs.Nodup) : xs.sym2.Nodup := by
...
+
+theorem map_mk_disjoint_sym2 (x : α) (xs : List α) (h : x ∉ xs) :
+    (map (fun y ↦ s(x, y)) xs).Disjoint xs.sym2 := by
+  induction xs with
+  | nil => simp
+  | cons x' xs ih =>
+    simp only [mem_cons, not_or] at h
+    rw [List.sym2, map_cons, map_cons, disjoint_cons_left, disjoint_append_right,
+      disjoint_cons_right]
+    refine ⟨?_, ⟨⟨?_, ?_⟩, ?_⟩⟩
```
Comment (YaelDillies) [inline Mathlib/Data/List/Sym.lean:145]:
```suggestion
    refine ⟨?_, ⟨?_, ?_⟩, ?_⟩
```
?

**Review by YaelDillies — APPROVED** [review]
Otherwise looks good

File: Mathlib/Data/List/Sym.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Data/List/Sym.lean]: Yes, done.

**Review by YaelDillies — APPROVED** [review]
Nice that you finally cracked the `dedup_sym2` nut!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**